Makefile.toml

```diff
@@ -463,6 +463,18 @@ fi
 '''
 ]
 
+[tasks.check-golangci-lint]
```

Comment:
If we add this as a dependency of `check-lints` then we wouldn't need the separate workflow since `cargo make check-lints` is covered in the `build` workflow.

Reply:
But then we also can't only have it run when *.go files have changed.

Reply:
I opted out of adding this as a dependency for `check-lints` since it will break everyone's build who don't already have Go and/or `golangci-lint` installed.

But, if we want to make that change and enforce these checks on every build, I'm all for it!

Reply:
Sure - If the prior-art is to do the lints on each run/build, I think that's fair.

marking this as draft to go back and iterate!

Reply:
> Does the lint checker take a long time? I think if it's very fast then I don't think it's a big issue. We do shellcheck for every PR right now as well.

It doesn't, but I would like to see that shellcheck and other checks broken out in our Actions so they are not run on every run. They don't individually take long, but right now we are running all of them for every variant for every PR. granted it's not a huge savings, but added up over time it would be a lot more efficient if we only ran checks when needed and only for things that have changed since the last time they were checked.

Reply:
> It doesn't, but I would like to see that shellcheck and other checks broken out in our Actions so they are not run on every run.

I wouldn't want to see this breakout take place without at least quantifying the amount of time being spent of the checks. If it's, e.g. 1% of the actions run, then I would prefer to run them for every variant/arch/build